### PR TITLE
Add user guide deploying ESP on a kubernetes cluster.

### PR DESCRIPTION
**Pull request by wlu2016**

### CODE REVIEW

**Comment by elibixby** [comment]
I'm not sure this deserves it's own folder. What are your thoughts on renaming GKE to k8s and then making a note about how if you're not on GKE you need to provide a service account?

**Comment by wlu2016** [comment]
good question. I thought about this too. It seems easier to follow the instructions by separating them to different folders. Otherwise we would add a lot of "if" in readme to distinguish these two cases. Also note we are using a separate script "start_esp.py" for k8s deployment.

gke/gce seems align well to cover GCP in this directory, with k8s to cover general platforms. Note k8s instruction with service account can also be used to on gke deployment.

**Comment by elibixby** [comment]
The only if we would need is "if you are running kubernetes not on GKE then do this service account stuff"

The rest of it should be identical right?

**Comment by wlu2016** [comment]
not exactly, we still need add that they need to remove "usr/sbin/start_esp.py" and remove "-k xxx".

Another point to keep this as a separate folder is that k8s user guide is support to be platform orthogonal. It feels a little strange to me if we need to add gke exception in the user guide. How do you think?

**Comment by elibixby** [comment]
Question: Why not create an environment variable pointing to the secret file in the k8s case and then you can use the same script for both? (As the service account will be picked up by ADC)?

**Comment by wlu2016** [comment]
Done merging k8s and gke doc/samples. PTAL.

**Comment by wlu2016** [comment]
@elibixby PTAL

File: k8s/README.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+         "-s", "SERVICE_NAME",
+         "-v", "SERVICE_VERSION",
+         "-k", "/etc/nginx/creds/service-account-creds.json",  # not needed for GKE
+       ]
+   ```
+
+   Note you also need to change the service type from LoadBalancer to NodePort
+   if you use [MiniKube](http://kubernetes.io/docs/getting-started-guides/minikube/)
+
+2. [Create your service account credentials](https://cloud.google.com/storage/docs/authentication#generating-a-private-key)
```
Comment (elibixby) [inline k8s/README.md]: 2 and 3 Should be merged into a single Bullet point, and you should START that bullet point with "(Not necessary if you kubernetes cluster is on GKE)"

**Comment by wlu2016** [comment]
done, ptal

**Comment by wlu2016** [comment]
@elibixby PTAL

**Comment by elibixby** [comment]
LGTM

**Pull request merged**